### ISSUE TITLE
Label positioning for retina display

### ISSUE THREAD

**Issue by nickasd** [chartjs/chartjs-plugin-datalabels#70]
I noticed that when slowly rotating a pie chart the labels (which also keep moving along with the chart) are noticeably jumping from one position to the next. I figured out that this is because their positions are rounded to integers without accounting for displays with a high pixel ratio (such as a Mac with retina display) where a position of, say, 11.5 pixels on the canvas is still pixel aligned on the display. I tried modifying the `drawText` function in `label.js` like this:
```
ctx.fillText(
	lines[i],
	Math.round(x * window.devicePixelRatio) / window.devicePixelRatio,
	Math.round(y * window.devicePixelRatio) / window.devicePixelRatio,
	Math.round(rect.w));
```
and the animation now looks nice. I'm not an expert in this matter though so I would be happy if someone could review this proposal.

**Comment by simonbrunel** [comment]
Positions are rounded just before being rendered to keep the labels sharp (mainly the frame borders) but I agree that it makes animations a bit jerky, especially if the `devicePixelRatio` > 1. Can you share a jsfiddle with the animation you are experiencing rendering issues?

**Comment by nickasd** [comment]
I agree that usually you would want to round the positions, and what I suggest would only optimise drawing on high resolution displays while still keeping the labels sharp.
Here is the jsfiddle: http://jsfiddle.net/vz42ogam/20/

**Comment by simonbrunel** [comment]
Fixed in ab410dd1b38b8791451731715980d50c7b44e9ff and will be available in [v0.4.0](/chartjs/chartjs-plugin-datalabels/milestone/4):

![retina](https://user-images.githubusercontent.com/3874900/43034559-ed8836f8-8cde-11e8-8785-705eecaa3109.gif)
